app/code/core/Mage/Api/Model/Resource/Role.php

```diff
@@ -44,27 +44,21 @@ protected function _construct()
     }
 
     /**
-     * Action before save
-     *
-     * @param Mage_Core_Model_Abstract $object
-     * @return $this
+     * @inheritDoc
      */
-    protected function _beforeSave(Mage_Core_Model_Abstract $object)
+    protected function _beforeSave(Mage_Core_Model_Abstract $role)
     {
-        if (!$object->getId()) {
-            $object->setCreated(now());
+        if ($role instanceof Mage_Admin_Model_Role) {
+            if (!$role->getId()) {
+                $role->setCreated(now());
+            }
+            $role->setModified(now());
         }
-        $object->setModified(now());
-        return $this;
+        return parent::_beforeSave($role);
```

Comment:
the call of the parent method is also new if I see this correctly.

Reply:
Also just for inherited methods from `Mage_Core_Model_Resource_Db_Abstract` ... just for code consistency. (parent just returns `$this`)

Reply:
Reopend below ...